languages/swift/exercises/concept/structs-and-classes/.meta/Example.swift

```diff
@@ -0,0 +1,75 @@
+struct Position {
+  var x = 0
+  var y = 0
+
+  mutating func moveTo(newX: Int, newY: Int) {
+    x = newX
+    y = newY
+  }
+}
+
+struct Size {
+  var width = 80
+  var height = 60
+
+  mutating func resize(newWidth: Int, newHeight: Int) {
+    width = newWidth
+    height = newHeight
+  }
+}
+
+class Window {
+  var title = "New Window"
+  let screenSize = Size(width: 800, height: 600)
+  var size = Size()
+  var position = Position()
+  var contents: String?
+
+  func move(to newPosition: Position) {
+    let minX = min(max(0, newPosition.x), screenSize.width - size.width)
+    let minY = min(max(0, newPosition.y), screenSize.height - size.height)
+    position.moveTo(newX: minX, newY: minY)
+  }
+  func move(deltaX: Int, deltaY: Int) {
+    let minX = min(max(0, position.x + deltaX), screenSize.width - size.width)
+    let minY = min(max(0, position.y + deltaY), screenSize.height - size.height)
+    position.moveTo(newX: minX, newY: minY)
+  }
+  func resize(to newSize: Size) {
+    let minW = min(max(1, newSize.width), screenSize.width - position.x)
+    let minH = min(max(1, newSize.height), screenSize.height - position.y)
+    size.resize(newWidth: minW, newHeight: minH)
+  }
+  func resize(deltaW: Int, deltaH: Int) {
+    let minW = min(max(1, size.width + deltaW), screenSize.width - position.x)
+    let minH = min(max(1, size.height + deltaH), screenSize.height - position.y)
+    size.resize(newWidth: minW, newHeight: minH)
+  }
```

Comment:
It looks like method overloading is used here. Are students familiar with that? Is that the reason why the functions are duplicated?

Reply:
There also is a lot of duplication that might not necessarily be a lot of fun to solve. Is there perhaps a way to cut down on this duplication?

Reply:
I'm sure I had a reason to add these when I wrote them, but I don't see much benefit to doubling them. I'll remove one set from the move and resize tasks.